### PR TITLE
restructure doc files for readTheDocs

### PR DESCRIPTION
**Pull request by nikita-fuchs**
currently, the readTheDocs documentation at https://aeternity-sophia.readthedocs.io/ points to my WIP restructuring, it should be fine though now to merge it to the official doc. content wise, I left the "big" `sophia.md` in place just as an internal reference. Regarding the `protected calls`: In the final text, there is a note about it not being implemented yet. as soon as things are merged, I will point the readthedocs link to the official sophia repo, as it should be.

### CODE REVIEW

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:11]: No... Again, this is not right. This is the Lima branch, this isn't implemented on Lima.

**Review by hanssv — CHANGES_REQUESTED** [review]
The Lima branch has documentation of things implemented in Lima. Period. 

Either merge this to master or remove the things not implemented in Lima.

File: docs/sophia.md
Comment (nikita-fuchs) [reply docs/sophia.md:11]: this file was for internal reference only anyway.

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:32]: remove

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:41]: remove

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:556]: remove

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:222]: keep the old version

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:698]: Remove this example doesn't make sense on lima.

**Review by hanssv — COMMENTED** [review]
Marked up the stuff that should and shouldn't change.

**Comment by nikita-fuchs** [comment]
can you check, @hanssv ?

File: docs/docs/contracts.md
Comment (hanssv) [inline docs/docs/contracts.md:145]: AENS.update should be in this list eventually, but no hurry it isn't released yet...

File: docs/docs/contracts.md
Comment (hanssv) [inline docs/docs/contracts.md:531]: Here a warning that Auth.tx isn't available until Iris is is missing.

File: docs/docs/sophia_stdlib.md
Comment (hanssv) [inline docs/docs/sophia_stdlib.md:543]: This isn't available until Iris.

File: docs/docs/sophia_stdlib.md
Comment (hanssv) [inline docs/docs/sophia_stdlib.md:1233]: A lot of this isn't available until Iris

File: docs/docs/sophia_stdlib.md
Comment (hanssv) [inline docs/docs/sophia_stdlib.md:1612]: This library isn't available until Iris

File: docs/sophia.md
Comment (hanssv) [inline docs/sophia.md:571]: Maybe I missed this one, but nevertheless, Auth.tx isn't in iris, so neither should its documentation.

File: docs/sophia_stdlib.md
Comment (hanssv) [inline docs/sophia_stdlib.md:17]: Iris only, shouldn't be in this branch.

File: docs/sophia.md
Comment (nikita-fuchs) [reply docs/sophia.md:571]: this file shall be deleted after the merge, it's just the big template for creating the separate files.

File: docs/docs/contracts.md
Comment (nikita-fuchs) [reply docs/docs/contracts.md:531]: done.

File: docs/docs/sophia_stdlib.md
Comment (nikita-fuchs) [reply docs/docs/sophia_stdlib.md:543]: marked.

File: docs/docs/sophia_stdlib.md
Comment (nikita-fuchs) [reply docs/docs/sophia_stdlib.md:1233]: marked.

File: docs/docs/sophia_stdlib.md
Comment (nikita-fuchs) [reply docs/docs/sophia_stdlib.md:1612]: marked.

File: docs/sophia_stdlib.md
Comment (nikita-fuchs) [reply docs/sophia_stdlib.md:17]: Removed.

**Comment by nikita-fuchs** [comment]
Did I miss anything, @hanssv ?

File: docs/docs/contracts.md
Comment (hanssv) [reply docs/docs/contracts.md:534]: Only Auth.tx is coming in Iris, the rest is already there. But there are no users of this, so nevermind 😢

**Review by hanssv — COMMENTED** [review]
You comment "marked" to a lot of things, but I see no marks @nikita-fuchs ?

File: docs/sophia.md
Comment (radrow) [inline docs/sophia.md:82]: Maybe worth noting "local mutable" by the way
```suggestion
- A contract may define a type `state` encapsulating its local mutable
```

File: docs/docs/contracts.md
Comment (marc0olo) [reply docs/docs/contracts.md:534]: @hanssv I am planning to test generalized accounts and provide some examples in the near future ;-) ... but I don't know when I will find the time to do so. will definitely play around with it because we also want to support the feature in the java sdk

**Comment by nikita-fuchs** [comment]
hello everyone, I'm not aware exactly how much the documentation has progressed internally already, but as this is what the public is currently served through readTheDocs, could we please merge this restructuring of the files and continue working on these documents here so things don't drift apart too far? If we wait too long getting this one here up to date will be a bit tedious I fear.

**Comment by omar-saadoun** [comment]
Hey guys @hanssv @radrow any feedback here ?

**Comment by hanssv** [comment]
Until the things that @nikita-fuchs commented as "marked" above is actually fixed I see no reason to merge this. But I really don't care, I know how the compiler works 😛 Also we are quickly approaching Iris, so it would probably make sense to prepare properly for that. I assume there is a plan for that?

I'll leave it to @radrow and @nikita-fuchs to untangle the mess they've created. Unfortunately you can't unrequest changes without actually accepting the PR so lets fix things first.

**Pull request closed without merging**
